src/App.tsx

```diff
@@ -15,47 +16,145 @@ export const goodsFromServer = [
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort(
+    (
+      wordOne, wordTwo,
+    ) => getSortChoice(
+      wordOne, wordTwo, sortType,
+    ),
+  );
+
+  if (isReversed === true) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [stateArray, setStateArray] = useState({
+    isReversed: false,
+    sortType: 0,
+  });
+
+  function sortByAlphabet() {
+    setStateArray({
+      isReversed: false,
+      sortType: SortType.ALPHABET,
+    });
+  }
+
+  function sortByLength() {
+    setStateArray({
+      isReversed: false,
+      sortType: SortType.LENGTH,
+    });
+  }
+
+  function reverse() {
+    setStateArray(prevState => ({
+      isReversed: !prevState.isReversed, sortType: prevState.sortType,
+    }));
+  }
+
+  function reset() {
+    setStateArray({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  }
+
+  const sort = stateArray.sortType;
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames('button', {
+            'is-info is-light': sort !== SortType.ALPHABET,
+            'is-info': sort === SortType.ALPHABET,
+          })}
+          onClick={sortByAlphabet}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames('button', {
+            'is-success is-light': sort !== SortType.LENGTH,
+            'is-success': sort === SortType.LENGTH,
+          })}
+          onClick={sortByLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames('button', {
+            'is-warning is-light': stateArray.isReversed !== true,
+            'is-warning': stateArray.isReversed === true,
+          })}
+          onClick={reverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(stateArray.sortType !== SortType.NONE
+          || stateArray.isReversed === true) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={reset}
+          >
+            Reset
+          </button>
+        )}
+
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {getReorderedGoods(
+            goodsFromServer, stateArray.sortType, stateArray.isReversed,
+          ).map(
```

Comment:
better to move in variable result of getReorderedGoods funcion
`const goods = getReorderedGoods(goodsFromServer, { isVisible, sortType });`

```suggestion
          {goods.map(
```